File: leetcode/977. Squares of a Sorted Array/977_squares_of_a_sorted_array.cs
public class Solution {
    public int[] SortedSquares(int[] nums) {
        int left = 0;
        int right = nums.Length - 1;
        int cur_index = right;
        int[] result = new int[nums.Length];
        
        while(left <= right)
        {
            int leftValue = (int)Math.Pow(nums[left],2);
            int rightValue = (int)Math.Pow(nums[right],2);
            if(leftValue < rightValue)
            {
                result[cur_index] = rightValue;
                cur_index--;
                right--;
            }
            else
            {
                result[cur_index] = leftValue;
                cur_index--;
                left++;
            }
        }
        return result;
    }
}

File: leetcode/977. Squares of a Sorted Array/977_squares_of_a_sorted_array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int left = 0;
        int right = nums.size()-1;
        int cur_index = right;
        vector<int> result(nums.size());
        
        while(left <= right){
            int leftValue = pow(nums[left], 2);
            int rightValue = pow(nums[right],2);
            if(leftValue < rightValue){
                result[cur_index] = rightValue;
                cur_index--;
                right--;
            }
            else{
                result[cur_index] = leftValue;
                cur_index--;
                left++;
            }
        }
        return result;
    }
};
